File: go/src/leetcode/offer/Offer64.go
package offer

/**
 * @author wujingxinit@outlook.com
 * @date 2023/1/4 02:54
 */

func sumNums(n int) int {
	ans := 0
	var dfs func(int)
	dfs = func(n int) {
		if n <= 0 {
			return
		}
		ans += n
		dfs(n - 1)
	}
	dfs(n)
	return ans
}

File: java/src/leetcode/offer/Offer64.java
package leetcode.offer;

/**
 * @author wujingxinit@outlook.com
 * @date 2023/9/13 22:21
 */
public class Offer64 {
    int res = 0;
    public int sumNums(int n) {
        boolean x = n > 1 && sumNums(n - 1) > 0;
        res += n;
        return res;
    }
}
